src/app/containers/Ad/index.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useContext } from 'react';
+import { RequestContext } from '../../contexts/RequestContext';
+import { ServiceContext } from '../../contexts/ServiceContext';
+import { GridItemConstrainedLargeWithAutoMargin } from '#lib/styledGrid';
+import useToggle from '../Toggle/useToggle';
+
+const constructAdJsonData = ({ id, pageType, service }) => {
+  const data = {
+    targeting: {
+      slot: 'mpu',
+      asset_type: pageType,
+      story_id: id,
+      channel: service,
+      sectn: service,
+      subject: '',
+    },
+  };
+
+  return data.toString();
+};
+
+const ampAdProps = ({ id, pageType, service }) => ({
+  type: 'doubleclick',
+  width: '300',
+  height: '250',
+  'data-slot': '/4817/bbccom.live.site.amp.news',
+  'data-amp-slot-index': '0',
+  'data-a4a-upgrade-type': 'amp-ad-network-doubleclick-impl',
+  json: constructAdJsonData({ id, pageType, service }),
+});
+
+const AdContainer = () => {
+  const { id, platform, pageType } = useContext(RequestContext);
+  const { service } = useContext(ServiceContext);
+  const { enabled } = useToggle('ampAds');
+
+  if (enabled && (platform === 'amp' || platform === 'canonical')) {
+    return (
+      <GridItemConstrainedLargeWithAutoMargin>
```

Comment:
Depending on the add requirements we may want to pull this out into a separate component as the UI may be different between the different slot types. This way we can have one container handling the business logic for all ads and then multiple components for the different ad types and their UI elements.